src/store/modules/order/actions.ts

```diff
@@ -0,0 +1,252 @@
+        orders.map((order: any) => {
+          order.doclist.docs.map((item: any) => {
+            orderItemsList.push(`${item.orderId}_${item.orderItemSeqId}`)
+            if(item.productId) productIds.add(item.productId)
+          })
+        })
+        productIds = [...productIds]
+
+        // Added check as we are fetching the facets only on first request call and do not fetch facets information on infinite scroll
+        if(params?.fetchFacets) {
+          const originFacilities = resp.data.facets?.facilityNameFacet?.buckets.map((bucket: any) => bucket.val)
+          const destinationFacilities = resp.data.facets?.orderFacilityNameFacet?.buckets.map((bucket: any) => bucket.val)
+          const productStores = resp.data.facets?.productStoreIdFacet?.buckets.map((bucket: any) => bucket.val)
+          const carriers = resp.data.facets?.carrierPartyIdFacet?.buckets.map((bucket: any) => bucket.val)
+          const shipmentMethodTypeIds = resp.data.facets?.shipmentMethodTypeIdFacet?.buckets.map((bucket: any) => bucket.val)
+          const statuses = resp.data.facets?.orderStatusDescFacet?.buckets.map((bucket: any) => bucket.val)
+
+          commit(types.ORDER_PRODUCT_STORE_OPTIONS_UPDATED, productStores);
+          commit(types.ORDER_ORIGIN_FACILITY_OPTIONS_UPDATED, originFacilities);
+          commit(types.ORDER_DESTINATION_FACILITY_OPTIONS_UPDATED, destinationFacilities);
+          commit(types.ORDER_CARRIERS_OPTIONS_UPDATED, carriers);
+          commit(types.ORDER_SHIPMENT_METHODS_OPTIONS_UPDATED, shipmentMethodTypeIds);
+          commit(types.ORDER_STATUS_OPTIONS_UPDATED, statuses);
+        }
+
+
+        const orderItemStats = await OrderService.fetchOrderItemStats(orderItemsList);
+        orders.map((order: any) => {
+          let totalOrdered = 0, totalShipped = 0, totalReceived = 0;
+          order.doclist.docs.map((item: any) => {
+            if(orderItemStats[`${item.orderId}_${item.orderItemSeqId}`]) {
+              item["shippedQty"] = orderItemStats[`${item.orderId}_${item.orderItemSeqId}`].shippedQty
+              item["receivedQty"] = orderItemStats[`${item.orderId}_${item.orderItemSeqId}`].receivedQty
+            }
+            if(item.quantity) totalOrdered = totalOrdered + item.quantity            
+            if(item.shippedQty) totalShipped = totalShipped + item.shippedQty            
+            if(item.receivedQty) totalReceived = totalReceived + item.receivedQty            
+          })
+
+          order["totalOrdered"] = totalOrdered
+          order["totalShipped"] = totalShipped
+          order["totalReceived"] = totalReceived
+        })
+
+        if (query.json.params.start && query.json.params.start > 0) cachedOrders = cachedOrders.concat(orders)
+        else cachedOrders = orders
+        await this.dispatch("product/fetchProducts", { productIds })
+        await this.dispatch("util/fetchShipmentMethodTypeDesc")
+      } else {
+        showToast(translate("Failed to fetch orders"));
```

Comment:
This toast should either be removed (if we have empty state in place) or should be moved to the catch part.